wandb/apis/public.py

```diff
@@ -4441,3 +4844,128 @@ def convert_objects(self):
 
     def __repr__(self):
         return "<ArtifactFiles {} ({})>".format("/".join(self.artifact.path), len(self))
+
+
+class Job(Media):
```

Comment:
Why is this a Media subclass? can you elaborate here?

Reply:
the ownership pattern here is a bit inverted w.r.t Media and Artifacts. an Artifact can have an entry that describes a Media object. However in this implementation, the constructor fetches an artifact. I think looking at how the Video media type works for example would be more akin to the desired pattern for media types

Reply:
the `from_json` method does all the artifact member reading before passing to an artifact-agnostic constructor